logictest: avoid parallelizing multiregion logic tests

otan opened the pull request:
Starting multi-node servers for multi-region is expensive, especially in
parallel. As such, avoid parallelising logic tests with multiple
servers.

Resolves #69272
Resolves #69276 

Release justification: testing only change

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/69342)
<!-- Reviewable:end -->

otan committed:
logictest: avoid parallelizing multiregion logic tests

Starting multi-node servers for multi-region is expensive, especially in
parallel. In CI, we use `-p 16`, which parallelizes this 16 ways which
is really intensive on the CPU, leading to weird gossip issues.
As such, avoid parallelising logic tests with multiple servers.

Release justification: testing only change

Release note: None

otan committed:
Revert "logictest: bump SET CLUSTER TIMEOUT wait setting for tests"

This reverts commit 20442b6a22e13a265b7767e6ef60787de65c6733.

Release justification: testing only change

Release note: None

andreimatei reviewed (commented):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

I'd be good for posterity to record in the commit message the `-p` setting currently used by CI (cause like if CI would pass `-p 1`, I think we'd probably be able to parallelize 16-ways or something inside this logictest package).

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/69342#-:-Mhy4hEKBk3F1NsQmTz4:bsrh6r0)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @arulajmani)



<!-- Sent from Reviewable.io -->

otan committed:
Revert "logictest: skip > 3 node logic tests in bazel"

This reverts commit be67bba2affab56ee42805cb0ce91d6977e0c287.

Release justification: Testing only change

Release note: None

RaduBerinde commented:
This change really slowed down the `execbuilder` tests. Before:
`ok  	github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder	18.511s`

After:
`ok  	github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder	41.798s`

I think it's mostly due to the `5node` config. Could we turn up the threshold to 5 instead of 3?

otan commented:
#72789
